### PR TITLE
add checks for hasOwnProperty and used forEach where useful

### PR DESCRIPTION
**Pull request by srir**
when Object.prototype is modified (or Array.prototype, for that matter), we can no longer iterate over the keys (or elements, respectively) using `for (var ... in ...)`. Added a hasOwnProperty check for object key iteration, and used `.forEach` for array element iteration instead.

### CODE REVIEW

File: lib/js/emojione.js [hunk 1]
```diff
@@ -343,9 +343,10 @@
     // needs to be improved into one big replacement instead, for performance reasons
     ns.toShort = function(str) { // this is really just unicodeToShortname() but I opted for the shorthand name to match toImage()
         for (var shortcode in ns.emojioneList) {
```
Comment (pspeter3) [inline lib/js/emojione.js:345]: As a note, `for ... in ...` is bad for performance. Using `Object.keys` and then doing a reverse for loop is preferable

Comment (srir) [inline lib/js/emojione.js:345]: i agree! that's for a second pass - this is basically so i can get actual, working (non-buggy) code in as a submodule.

what do you mean 'reverse for loop'?

Comment (pspeter3) [inline lib/js/emojione.js:345]:
``` ts
function forEach<T>(obj: {[key: string]: T}, callback: (item: T) => any): void {
    var keys = Object.keys(obj);
    for(var i = keys.length - 1; i >= 0; i--) {
          fn(obj[keys[i]]);
    }
}
```

Comment (pspeter3) [inline lib/js/emojione.js:345]: Basically checking length is apparently expensive so you need to cache it in a local variable or loop backwards. Also `obj[key] !== undefined` is faster than `obj.hasOwnProperty(key)`

Comment (vsiao) [inline lib/js/emojione.js:345]: I'd say clarity > optimization in an emoji library

Comment (pspeter3) [inline lib/js/emojione.js:345]: That's also reasonable. I find both clear

**Pull request merged**